src/FixedDataTable.js

```diff
@@ -474,74 +509,48 @@ var FixedDataTable = React.createClass({
       );
     }
 
-    var maxScrollY = this.state.maxScrollY;
```

Comment:
moved to verticalHeightsSelector and untangled all this.  

The logic in the selector is now linear, e.g. A is used to compute B and if X, B & C are used to compute D,
instead of the tangle it can be here, e.g. Q is used to compute R, and if X, R & W are used to compute Q.